test/bun.js/fetch.test.js

```diff
@@ -8,6 +8,66 @@ const exampleFixture = fs.readFileSync(
   "utf8",
 );
 
+let server ;
+beforeAll(() => {
+  server = Bun.serve({
+    async fetch(){
+      await Bun.sleep(2000);
```

Comment:
There's a lot more we should test here

- what happens when you use ReadableStream and it's aborted? 
- does request.signal.addListener work? 
- what happens when you call request.signal.abort() but the request has already been responded to (does it crash?) 
- what happens when you use ReadableStream type: "direct" and it's aborted? 
- what happens when you use ReadableStream, you call stream.cancel() and then request.signal? Does it crash? 

Later we should make Bun.spawn work with AbortSignal as well

Reply:
Working in the `Request` object signal and next will I can make `Bun.spawn`, I will write those tests ASAP, you are right my tests are not enough, we need a more complete test suit.

Request object will affect fetch and maybe `Bun.server` (I think Bun.server should not be affected)?